### PR TITLE
add obfuscation option to aar

### PR DESCRIPTION
**Pull request by xster**

### CODE REVIEW

**Review by jonahwilliams — COMMENTED** [review]
on L109 a BuildInfo is constructed from mode/flavor and tree shake icons. That needs to be updated to do everything that https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L543 does

File: packages/flutter_tools/lib/src/commands/build_aar.dart [hunk 1]
```diff
@@ -102,14 +104,17 @@ class BuildAarCommand extends BuildSubCommand {
       ? stringArg('build-number')
       : '1.0';
 
-    for (final String buildMode in const <String>['debug', 'profile', 'release']) {
+    const Map<String, BuildInfo> buildModes = {'debug': BuildInfo.debug, 'profile': BuildInfo.profile, 'release': BuildInfo.release};
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/build_aar.dart:107]: You're still using default values here, you'll need to explicitly construct a build info with all of the parameters set that you care about: https://github.com/flutter/flutter/blob/dd88b20435c1eb9b837bb4aa8b0a84aa0825530d/packages/flutter_tools/lib/src/build_info.dart#L14

Comment (xster) [reply packages/flutter_tools/lib/src/commands/build_aar.dart:107]: seems like that's what build_ios_framework is doing too. Does that one need refactoring too?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/build_aar.dart:107]: 🤷‍♂️

**Pull request closed without merging**